test/src/test/kotlin/graphql/nadel/tests/hooks/execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf.kt

```diff
@@ -0,0 +1,94 @@
+package graphql.nadel.tests.hooks
+
+import graphql.ExecutionResult
+import graphql.ExecutionResultImpl
+import graphql.execution.AbortExecutionException
+import graphql.execution.instrumentation.InstrumentationContext
+import graphql.execution.instrumentation.InstrumentationState
+import graphql.nadel.Nadel
+import graphql.nadel.instrumentation.NadelInstrumentation
+import graphql.nadel.instrumentation.parameters.NadelInstrumentationCreateStateParameters
+import graphql.nadel.instrumentation.parameters.NadelInstrumentationExecuteOperationParameters
+import graphql.nadel.instrumentation.parameters.NadelInstrumentationQueryExecutionParameters
+import graphql.nadel.tests.EngineTestHook
+import graphql.nadel.tests.KeepHook
+import graphql.nadel.tests.NadelEngineType
+import graphql.nadel.tests.util.data
+import graphql.nadel.tests.util.errors
+import graphql.nadel.tests.util.extensions
+import graphql.nadel.tests.util.getToString
+import graphql.nadel.tests.util.message
+import strikt.api.expectThat
+import strikt.assertions.get
+import strikt.assertions.isEqualTo
+import strikt.assertions.isNotNull
+import strikt.assertions.isNull
+import strikt.assertions.single
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletableFuture.completedFuture
+
+@KeepHook
+class `execution-is-aborted-when-begin-execute-completes-exceptionally-inside-cf` : EngineTestHook {
+    var instrumentationParams: NadelInstrumentationExecuteOperationParameters? = null
+    var resultBeforeFinalInstrumentation: ExecutionResult? = null
+
+    override fun makeNadel(engineType: NadelEngineType, builder: Nadel.Builder): Nadel.Builder {
+        return builder
+            .instrumentation(object : NadelInstrumentation {
+                override fun createState(parameters: NadelInstrumentationCreateStateParameters): InstrumentationState {
+                    return object : InstrumentationState {
+                        override fun toString(): String {
+                            return "so-annoying"
```

Comment:
hm, why is this annoying?

Reply:
Ah it's a copy paste from very similar test, I think I had issues with assertions when I was writing the test.